src/shared/components/case-editor/services/cases.service.ts

```diff
@@ -217,15 +219,12 @@ export class CasesService {
     if (!eventTrigger.wizard_pages) {
       eventTrigger.wizard_pages = [];
     }
-    /* FIXME: find a better place for this code */
+
     eventTrigger.wizard_pages.forEach((wizardPage: WizardPage) => {
       wizardPage.parsedShowCondition = new ShowCondition(wizardPage.show_condition);
-      let orderedWPFields = this.orderService.sort(wizardPage.wizard_page_fields);
-      wizardPage.case_fields = orderedWPFields.map((wizardField: WizardPageField) => {
-        let case_field = eventTrigger.case_fields.find(cf => cf.id === wizardField.case_field_id);
-        case_field.wizardProps = wizardField;
-        return case_field;
-      });
+      wizardPage.case_fields = this.orderService
+        .deepSort(this.wizardPageFieldToCaseFieldMapper.mapAll(wizardPage.wizard_page_fields, eventTrigger.case_fields));
```

Comment:
shall we rename `wizardPageFieldToCaseFieldMapper` to a shorter `wizardFieldToCaseFieldMapper`?

Reply:
No. If is a fully qualified name and it was generated by the IDE. I know it is long, but it is telling what type of object we are mapping to which one.